Banno/ux-lint
# Add linting for HTML files

fixes #10

Uses [HTMLHint](https://github.com/yaniswang/HTMLHint) for the basic checking, plus adds a few custom rules. See the `config/htmlhint.hjson` file for details of what is checked.

cc @Banno/ux

## How to Test

* `npm run lint` and `npm test` should pass.
* You can run it against our projects to see the output:
  * `node cli.js /path/to/banno-web/src/**/*.html`
  * `node cli.js /path/to/platform-ux/src/index.html /path/to/platform-ux/src/*/*.html`

## Notes

It's not possible to whitelist tags the way you can for attributes. So we can whitelist SVG attributes like `viewBox` but not SVG tags like `clipPath`. This has only been used in 1 file so far though.

## Discussion Points

* I set `space-tab-mixed-disabled` to tabs, to be consistent with our JS guidelines. But we don't really use tabs for HTML files -- would it be better to be inconsistent here and potentially more confusing?
* Is the HTMLHint library good enough? It looks like not much development is going on with it, but it works pretty well currently, and for any changes we should be able to fork and modify it w/o too much trouble.
